rust/main/hyperlane-core/Cargo.toml

```diff
@@ -17,8 +17,10 @@ bigdecimal.workspace = true
 borsh.workspace = true
 bs58.workspace = true
 bytes = { workspace = true, features = ["serde"] }
+cainome = { workspace = true, optional = true }
 config = { workspace = true, optional = true }
 convert_case.workspace = true
+cosmrs = { workspace = true, optional = true }
```

Comment:
we're aiming to remove all chain-specific dependencies from `hyperlane-core` since we're running into increasingly more dependency conflicts as we add new VMs. The logic specific to these crates (`cainome`, `cosmrs`, `starknet`) looks to only be conversion-related, which can live in e.g. the `hyperlane-*` crate specific to that VM. I believe this is why we never actually implemented the `From` trait for cosmos TxResponses and instead use `tx_response_to_outcome`; may need to do something similar for starknet

Reply:
makes sense, done